src/messaging/views.py

```diff
@@ -35,25 +35,27 @@ def websockets_notify_user(user):
 
 # Create your views here.
 
-class ReadMessageView(generics.UpdateAPIView):
+class ReadThreadView(generics.UpdateAPIView):
     """
-    View for marking a specified message as read
+    View for marking a specified thread as read
     """
-    serializer_class = MessageSerializer
+    serializer_class = OwnThreadSerializer
 
     def patch(self, request, *args, **kwargs):
-        message_id = int(self.kwargs['message_id'])
-        message = get_object_or_404(Message, id=message_id)
-
-        message.unread = False
+        thread_id = int(self.kwargs['thread_id'])
+        thread = get_object_or_404(Thread, id=thread_id)
+        unread_messages = Message.objects.filter(Q(thread=thread) & Q(unread=True))
 
-        message.save()
+        for message in unread_messages:
+            message.unread = False
+            message.save()
```

Comment:
this is a "n+1 query".  you can look up what it is and why it isn't great.

It honestly doesnt matter for our scale, but I think you would learn something by fixing it.  

use the `unread_messages` queryset and update them all at once